Sources/AppcuesKit/Presentation/Traits/Appcues/AppcuesTooltipTrait.swift

```diff
@@ -38,15 +45,20 @@ internal class AppcuesTooltipTrait: StepDecoratingTrait, WrapperCreatingTrait, P
     func createWrapper(around containerController: ExperienceContainerViewController) throws -> UIViewController {
         let dialogWrapperViewController = ExperienceWrapperViewController<TooltipWrapperView>(wrapping: containerController)
         dialogWrapperViewController.configureStyle(tooltipStyle)
+        dialogWrapperViewController.bodyView.pointerSize = hidePointer ? nil : pointerSize
+        if let preferredWidth = tooltipStyle?.width {
+            dialogWrapperViewController.bodyView.preferredWidth = preferredWidth
+        }
+        // Ensure content isnt hidden by the tooltip pointer
+        dialogWrapperViewController.bodyView.pointerInsetHandler = { insets in
+            containerController.additionalSafeAreaInsets = insets
+        }
```

Comment:
just for my own learning, curious how this part works

Reply:
The idea is that the space inside the pointer is part of the modal. This enabled a background image to fill the pointer space, and any scrolling content to scroll into the pointer space. 

L139 adds the height of the pointer to the view sizing. But, for example, if we have a top pointer, we don't want the content to start in the pointer space, so the right way to address that is with the safe area insets.

That leaves us in a place where the tooltip view needs to be able to modify the `additionalSafeAreaInsets` for the `containerController` based on where the pointer is determined to be positioned.  The challenge is the tooltip view doesn't know about the `containerController`. This closure bridges the gap by allowing the trait that creates the `ExperienceWrapperViewController` to provide the tooltip view a way how to modify the necessary insets (`didSet` on L30) when the new pointer position is determined (`setPointerInset()` on L275).

It's definitely convoluted, but it's the best approach I could think of.

Reply:
makes sense! thanks for the details. It does feel like a proper approach to share that information in a necessarily decoupled way.